### PR TITLE
Creating Surveys - Initial Commit:

### PR DESCRIPTION
**Pull request by Bnortron**
- Added "title", "description", "status" columns to Survey.java for table formatting
- @OneToMany mapping for the SurveyQuestion list in Survey.java
- Added createSurvey(), saveSurvey(), selectSurvey(), viewSurvey(), showSurvey(), and viewSurveys() to SurveyController.java class
- Added corresponding html files for the methods in SurveyController.java (index.html, createsurvey.html, selectsurvey.html, showsurvey.html, and viewsurveys.html)
- Added application.properties file for setting up H2 database (and also added a shortcut to h2-console in the index.html file)
- Changes to SurveyQuestion.java, MultipleChoiceQuestion.java, NumberQuestion.java, and TextQuestion.java for future implementation of adding a SurveyQuestion to an existing Survey (NOTE: At the moment, you will create a blank Survey with a title and a description, it gets saved in the SurveyRepository, then SurveyQuestions are added afterwards -- have not gotten to the adding questions part yet)
- QuestionType.java ENUM class added to help with mapping the different types of questions
- MultipleChoiceQuestionTest.java, NumberQuestionTest.java, and TextQuestionTest.java all modified to function with the new constructors

### CODE REVIEW

**Comment by Aleksandar-Veselinovic** [comment]
do we need all the new xml files or is it just to make things easier

**Comment by ljjb97** [comment]
This is missing tests we get those before we commit it to main?

**Comment by Bnortron** [comment]
> 

If we're all using Intellij then it's good to have them (especially jsLibraryMappings and sqldialects), I would only exclude them if someone is working in a different IDE like Eclipse. Merging with them included for now shouldn't cause any issues and we can remove later if there's any issues from TA (in the case they're not using Intellij).

**Comment by Bnortron** [comment]
> 

Didn't think about tests, that's my bad. They should be added before merging but not sure if I have enough time to get proper coverage before the demo tomorrow. Since Integration testing isn't an issue at this stage, we can still merge without issues and add unit tests at a later time. Either way is okay for me though, let me know what you think

**Comment by Aleksandar-Veselinovic** [comment]
> > 
> 
> If we're all using Intellij then it's good to have them (especially jsLibraryMappings and sqldialects), I would only exclude them if someone is working in a different IDE like Eclipse. Merging with them included for now shouldn't cause any issues and we can remove later if there's any issues from TA (in the case they're not using Intellij).

ok sounds good

**Comment by ljjb97** [comment]
I would really like to have tests before we merge. Since this merge also completes the survey controller issue I can write some tests for it before we merge it into main.

File: src/main/java/com/example/SurveyController.java [hunk 1]
```diff
@@ -1,7 +1,107 @@
...
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
 
-@RestController
+import java.util.ArrayList;
+import java.util.Optional;
+
+@Controller
+@RequestMapping("/")
```
Comment (Aleksandar-Veselinovic) [inline src/main/java/com/example/SurveyController.java:12]: should this be here

Comment (Bnortron) [reply src/main/java/com/example/SurveyController.java:12]: At this point no, each of the controller methods have a URL pattern specified anyways. Once the testing is complete we can remove it before merging.

Comment (Aleksandar-Veselinovic) [reply src/main/java/com/example/SurveyController.java:12]: ok

**Comment by ljjb97** [comment]
I added some initial tests for this but there could be more. Should be okay to merge with this but it probably should be reviewed by someone else as well

File: src/main/java/com/example/SurveyController.java [hunk 2]
```diff
@@ -1,7 +1,106 @@
...
+    }
+
+    /**
+     * Displays the single Survey chosen from the list of ALL Surveys
+     *
+     * @param surveyId the id of the Survey to view
+     * @param model
+     * @return
+     */
+    @GetMapping("/survey/view")
```
Comment (Aleksandar-Veselinovic) [inline src/main/java/com/example/SurveyController.java:59]: from what I saw we don't test this endpoint right

**Pull request merged**